File: dict.java
package langnotes;
import java.util.*;

public class dict {
    public static void main(String[] args) {
        HashMap<String, String> dinner = new HashMap<>();
        dinner.put("appetizer", "apple");
        dinner.put("beverage", "beer");
        dinner.put("course", "crawfish");
        dinner.put("desert", "donut");
        for(Iterator i = dinner.entrySet().iterator(); i.hasNext();) {
            Map.Entry j = (Map.Entry)i.next();
            System.out.printf("%s: %s\t", j.getKey(), j.getValue());
        }
        return;
    }
}

File: dict.cpp
#include <iostream>
#include <map>
using namespace std;
int main(int argc, const char * argv[]) {
    map<string, string> dinner = {
        {"appetizer", "apple"},
        {"beverage", "beer"},
        {"course", "crawfish"},
        {"desert", "donut"}
    };
    for(const auto& i: dinner) {
        string key = i.first;
        string value = i.second;
        cout << key << ":\t" << value << endl;
    }
    return 0;
}
